src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Builder/OpenTelemetryBuilderOtlpExporterExtensions.cs

```diff
@@ -0,0 +1,216 @@
+
+        return UseOtlpExporter(builder, name: null, configure: configure);
+    }
+
+    internal static IOpenTelemetryBuilder UseOtlpExporter(
+        this IOpenTelemetryBuilder builder,
+        IConfiguration configuration)
+    {
+        Guard.ThrowIfNull(configuration);
+
+        return UseOtlpExporter(builder, name: null, configuration: configuration);
+    }
+
+    internal static IOpenTelemetryBuilder UseOtlpExporter(
+        this IOpenTelemetryBuilder builder,
+        string? name = null,
+        IConfiguration? configuration = null,
+        Action<OtlpExporterBuilder>? configure = null)
+    {
+        Guard.ThrowIfNull(builder);
+
+        if (configuration != null && string.IsNullOrEmpty(name))
+        {
+            name = "otlp";
+        }
+
+        var otlpBuilder = new OtlpExporterBuilder(builder.Services, name, configuration);
+
+        configure?.Invoke(otlpBuilder);
+
+        UseOtlpExporterInternal(builder, name);
+
+        return builder;
+    }
+
+    private static void UseOtlpExporterInternal(IOpenTelemetryBuilder builder, string? name)
+    {
+        name ??= Options.DefaultName;
+
+        // Note: We automatically turn on signals for "UseOtlpExporter"
+        builder
+            .WithLogging()
+            .WithMetrics()
+            .WithTracing();
+
+        var services = builder.Services;
+
+        // Note: UseOtlpExporterRegistration is added to the service collection
+        // to detect repeated calls to "UseOtlpExporter" and to throw if
+        // "AddOtlpExporter" extensions are called
+        services.AddSingleton<UseOtlpExporterRegistration>();
+
+        services.RegisterOptionsFactory(configuration => new SdkLimitOptions(configuration));
+        services.RegisterOptionsFactory(configuration => new ExperimentalOptions(configuration));
+        services.RegisterOptionsFactory((sp, configuration, name) => new OtlpExporterBuilderOptions(
+            configuration,
+            sp.GetRequiredService<IOptionsMonitor<SdkLimitOptions>>().CurrentValue,
+            sp.GetRequiredService<IOptionsMonitor<ExperimentalOptions>>().CurrentValue,
+            /* Note: We allow LogRecordExportProcessorOptions,
+            MetricReaderOptions, & ActivityExportProcessorOptions to be null
+            because those only exist if the corresponding signal is turned on.
+            Currently this extension turns on all signals so they will always be
+            there but that may change in the future so it is handled
+            defensively. */
+            sp.GetService<IOptionsMonitor<LogRecordExportProcessorOptions>>()?.Get(name),
+            sp.GetService<IOptionsMonitor<MetricReaderOptions>>()?.Get(name),
+            sp.GetService<IOptionsMonitor<ActivityExportProcessorOptions>>()?.Get(name)));
+
+        services.ConfigureOpenTelemetryLoggerProvider(
+            (sp, logging) =>
+            {
+                var builderOptions = GetBuilderOptionsAndValidateRegistrations(sp, name);
```

Comment:
Could we do this once outside of the individual signal configuration? It looks like they all check the same thing and retrieve the same options object?

Reply:
Not really. We need the `IServiceProvider` and there isn't currently a way to get it outside of these provider configuration extensions. Could possibly add something, but probably needs to be its own effort and then we can clean this up.